src/middleware.rs

```diff
@@ -2,31 +2,76 @@ pub(crate) mod messages;
 
 mod request_id;
 
-use crate::{controller, generic_http::GenericRequest, har_builder::HarBuilder, SpeakeasySdk};
-use log::debug;
-use std::collections::HashMap;
+use crate::{
+    controller,
+    generic_http::{GenericRequest, GenericResponse},
+    har_builder::HarBuilder,
+    SpeakeasySdk,
+};
+use http::{header::InvalidHeaderValue, HeaderValue, Uri};
+use log::{debug, error};
+use once_cell::sync::Lazy;
+use speakeasy_protos::ingest::{ingest_service_client::IngestServiceClient, IngestRequest};
+use std::{collections::HashMap, str::FromStr};
+use thiserror::Error;
+
+use tonic03::{transport::Error as TonicError, Request};
 
 use self::messages::MiddlewareMessage;
 
+static SPEAKEASY_SERVER_SECURE: Lazy<bool> = Lazy::new(|| {
+    !matches!(
+        std::env::var("SPEAKEASY_SERVER_SECURE").as_deref(),
+        Ok("false")
+    )
+});
+
+static SPEAKEASY_SERVER_URL: Lazy<String> = Lazy::new(|| {
+    let domain = std::env::var("SPEAKEASY_SERVER_URL")
+        .unwrap_or_else(|_| "grpc.prod.speakeasyapi.dev:443".to_string());
+
+    if !domain.starts_with("http") {
+        if *SPEAKEASY_SERVER_SECURE {
```

Comment:
This was done so users could set the `SPEAKEASY_SERVER_URL` as both `grpc.prod.speakeasyapi.dev:443` or `https://grpc.prod.speakeasyapi.dev:443`. Without this check it would have to be `https://grpc.prod.speakeasyapi.dev:443` (with the scheme) I can get rid of it tho.

Reply:
right, I don't think it needs the `http://` or `https://` prefixed

Reply:
Oh I see, ya I had to add the scheme or the underlying http library (hyper) or the gRPC library tonic errors out